coderd/prometheusmetrics/prometheusmetrics_test.go

```diff
@@ -323,6 +349,33 @@ func TestWorkspaceLatestBuildStatuses(t *testing.T) {
 			codersdk.ProvisionerJobSucceeded: 3,
 			codersdk.ProvisionerJobRunning:   1,
 		},
+	}, {
+		Name: "MultipleWithDeleted",
+		Database: func() database.Store {
+			db, _ := dbtestutil.NewDB(t)
+			u := dbgen.User(t, db, database.User{})
+			org := dbgen.Organization(t, db, database.Organization{})
+			insertTemplates(t, db, u, org)
+			insertCanceled(t, db, u, org)
+			insertFailed(t, db, u, org)
+			insertSuccess(t, db, u, org)
+			insertRunning(t, db, u, org)
+
+			// Verify that deleted workspaces/builds are NOT counted in metrics.
+			n, err := cryptorand.Intn(5)
+			require.NoError(t, err)
+			for range 1 + n {
+				insertDeleted(t, db, u, org)
+			}
```

Comment:
Review: This fails in the old implementation, but I thought it best to formalize the assumption. Considering the old implementation did not test for this, it seems like validation for the new approach.